src/helpers/enrich-dataset.ts

```diff
@@ -0,0 +1,119 @@
+import { IItem } from '@esri/arcgis-rest-portal';
+import { DatasetResource, datasetToItem, datasetToContent, getProxyUrl, getContentSiteUrls, IHubRequestOptions, IModel, parseDatasetId } from '@esri/hub-common';
+import { isPage } from '@esri/hub-sites';
+import * as _ from 'lodash';
+const WFS_SERVER = 'WFSServer';
+const WMS_SERVER = 'WMSServer';
+
+type HubDataset = Record<string, any>;
+export const enrichDataset = (
+    dataset: HubDataset,
+    siteUrl: string,
+    portalUrl: string,
+    siteModel: IModel): HubDataset => {
+    // Download and Hub Links must be generated from Content
+    const content = datasetToContent({
+        id: dataset.id,
+        attributes: dataset
+    } as DatasetResource);
+
+    const { relative: relativePath } = getContentSiteUrls(content, siteModel);
+    const hubLandingPage = siteUrl.startsWith('https://') ? siteUrl + relativePath : `https://${siteUrl}${relativePath}`;
```

Comment:
From what I understand, the logic was implemented because sometimes  `siteUrl ` might show up sometimes with `https://` even though we use `request.hostname`. I suspect there was weird bug and this was the fix. I can investigate further.